File: programmers/js/level01/12940.js
// 최대공약수와 최소공배수
function solution(n, m) {
  const gcdNum = gcd(n, m);
  
  return [gcdNum, n * m / gcdNum];
}

function gcd(n, m) {
  let r = n % m;

  if (r === 0) {
    return m;
  } else {
    return gcd(m, r);
  }      
}

// function gcd(a, b) {
//   let r;

//   while ((a % b) > 0)  {
//     r = a % b;
//     a = b;
//     b = r;
//   }

//   return b;
// }

File: programmers/java/level01/12940.java
// 최대공약수와 최소공배수

class Solution {
    int gcd(int a, int b) {
        while (b != 0) {
            int r = a;
            a = b;
            b = r % a;
        }
        
        return a;
    }
    
    public int[] solution(int n, int m) {
        int gcdNum = gcd(n, m);
        
        return new int[]{ gcdNum, n * m / gcdNum };
    }
}